### PR TITLE
completed api-authentication

### PR DESCRIPTION
**Pull request by BrianBozi**
Screen Shot 2021-07-16 at 10 04 17 AM

### CODE REVIEW

File: api-authentication/server/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+.then(result => {
+  const [user] = result.rows
+  if(!user) {
+    throw new ClientError(404, 'Invalid Login')
+  }
+  console.log(result)
+  const {userId, hashedPassword} = user;
+  argon2
+  .verify(hashedPassword, password)
+    .then(isMacthing => {
```
Comment (BrettAlbright) [inline api-authentication/server/index.js]: Hey Brian, the only update here is that if it does not match you want to send back a 401 client error.

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
Great work, small update!

**Comment by BrianBozi** [comment]
Screen Shot 2021-07-18 at 11 01 33 PM

File: api-authentication/server/index.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+.then(result => {
+  const [user] = result.rows
+  if(!user) {
+    throw new ClientError(401, 'Invalid Login')
+  }
+  const {userId, hashedPassword} = user;
+  return argon2
+  .verify(hashedPassword, password)
+    .then(isMatching => {
+      console.log("does the password match?",isMatching)
```
Comment (scbowler) [inline api-authentication/server/index.js:70]: Remember to remove any console logs before submitting a PR

**Review by scbowler — APPROVED** [review]
Looks good! 👍 

1 FYI comment, no changes required

**Pull request merged**